cc/dual_net/lite_dual_net.cc

```diff
@@ -73,16 +75,30 @@ minigo::LiteDualNet::LiteDualNet(std::string graph_path)
 
   // Let's just use all the processors we can.
   interpreter_->SetNumThreads(get_nprocs());
+}
+
+void minigo::LiteDualNet::Reserve(int capacity) {
+  MG_CHECK(capacity > 0);
+  if (capacity <= batch_capacity_) {
+    return;
+  }
 
   // Initialize input.
   const auto& inputs = interpreter_->inputs();
   MG_CHECK(inputs.size() == 1);
   absl::string_view input_name = interpreter_->GetInputName(0);
   MG_CHECK(input_name == "pos_tensor");
 
+  // Resize input tensor to batch size.
```

Comment:
I think you can move most of the checks into the constructor.
The  only checks you need in `Reserve` are something like:

```c++
  MG_CHECK(interpreter_->ResizeInputTensor(
               inputs[0], {static_cast<int>(capacity), kN, kN,
                           DualNet::kNumStoneFeatures}) == kTfLiteOk);
  MG_CHECK(interpreter_->AllocateTensors() == kTfLiteOk);
  MG_CHECK(input_->dims->data[0] == capacity);
  MG_CHECK(policy_->dims->data[0] == capacity);
  MG_CHECK(value_->dims->data[0] == capacity);
```

We can probably get rid of the `data.raw` checks, since presumably the fact that `ResizeInputTensor` succeeded guarantees that they're non-null.

Reply:
No. Most of those checks are not necessary after changing the batch size.

The reason the width, height & depth checks were put in the constructor originally is to make sure the model being loaded matches expectations, e.g. you're not trying to load a 9x9 model into an engine compiled for 19x19 games.

Changing the batch size will not change the width, height or depth. Having those checks immediately after calling ResizeInputTensor suggests that's a possibility, which is incorrect.

Checking that the number of outputs a model has remains constant after calling ResizeInputTensor just doesn't make any sense.